module/os/linux/zfs/zfs_vfsops.c

```diff
@@ -1596,6 +1617,7 @@ zfs_preumount(struct super_block *sb)
 		    dmu_objset_pool(zfsvfs->z_os)), 0);
 		taskq_wait_outstanding(dsl_pool_zrele_taskq(
 		    dmu_objset_pool(zfsvfs->z_os)), 0);
+		generic_shutdown_super(sb);
```

Comment:
This currently happens in `zpl_kill_sb()->kill_anon_super()` after the call to `zfs_preumount()`.  So this this redundant and can be dropped.